executor/fktest/foreign_key_test.go

```diff
@@ -2072,3 +2072,150 @@ func TestForeignKeyOnInsertOnDuplicateUpdate(t *testing.T) {
+		},
+		// Test foreign key use index.
+		{
+			sql: "explain insert into t4 values (1)",
+			plan: "Insert_1 N/A root  N/A]\n" +
+				"[└─Foreign_Key_Check_3 0.00 root table:t3, index:idx foreign_key:fk, check_exist",
+		},
+		{
+			sql: "explain update t4 set id=id+1 where id = 1",
+			plan: "Update_4 N/A root  N/A]\n" +
+				"[└─IndexReader_7 10.00 root  index:IndexRangeScan_6]\n" +
+				"[  └─IndexRangeScan_6 10.00 cop[tikv] table:t4, index:idx_id(id) range:[1,1], keep order:false, stats:pseudo]\n" +
+				"[└─Foreign_Key_Check_8 0.00 root table:t3, index:idx foreign_key:fk, check_exist",
+		},
+		{
+			sql: "explain delete from t3 where id > 1",
+			plan: "Delete_4 N/A root  N/A]\n" +
+				"[└─IndexReader_7 3333.33 root  index:IndexRangeScan_6]\n" +
+				"[  └─IndexRangeScan_6 3333.33 cop[tikv] table:t3, index:idx(id) range:(1,+inf], keep order:false, stats:pseudo]\n" +
+				"[└─Foreign_Key_Check_8 0.00 root table:t4, index:idx_id foreign_key:fk, check_not_exist",
+		},
+		{
+			sql: "explain update t3 set id=id+1 where id = 1",
+			plan: "Update_3 N/A root  N/A]\n" +
+				"[└─Point_Get_1 1.00 root table:t3, index:idx(id) ]\n" +
+				"[└─Foreign_Key_Check_4 0.00 root table:t4, index:idx_id foreign_key:fk, check_not_exist",
+		},
+		{
+			sql:  "explain insert into t3 values (1)",
+			plan: "Insert_1 N/A root  N/A",
+		},
+		{
+			sql: "explain insert into t3 values (1) on duplicate key update id = 100",
+			plan: "Insert_1 N/A root  N/A]\n" +
+				"[└─Foreign_Key_Check_3 0.00 root table:t4, index:idx_id foreign_key:fk, check_not_exist",
+		},
+		// Test multi-foreign keys in on table.
+		{
+			sql: "explain insert into t6 values (1,1,1)",
+			plan: "Insert_1 N/A root  N/A]\n" +
+				"[└─Foreign_Key_Check_3 0.00 root table:t5 foreign_key:fk_1, check_exist]\n" +
+				"[└─Foreign_Key_Check_4 0.00 root table:t5, index:idx2 foreign_key:fk_2, check_exist]\n" +
+				"[└─Foreign_Key_Check_5 0.00 root table:t5, index:idx3 foreign_key:fk_3, check_exist",
+		},
+		{
+			sql: "explain update t6 set id=id+1, id3=id2+1 where id = 1",
+			plan: "Update_4 N/A root  N/A]\n" +
+				"[└─IndexLookUp_11 10.00 root  ]\n" +
+				"[  ├─IndexRangeScan_9(Build) 10.00 cop[tikv] table:t6, index:idx_id(id) range:[1,1], keep order:false, stats:pseudo]\n" +
+				"[  └─TableRowIDScan_10(Probe) 10.00 cop[tikv] table:t6 keep order:false, stats:pseudo]\n" +
+				"[└─Foreign_Key_Check_12 0.00 root table:t5 foreign_key:fk_1, check_exist]\n" +
+				"[└─Foreign_Key_Check_13 0.00 root table:t5, index:idx3 foreign_key:fk_3, check_exist",
+		},
+		{
+			sql: "explain delete from t5 where id > 1",
+			plan: "Delete_4 N/A root  N/A]\n" +
+				"[└─TableReader_7 3333.33 root  data:TableRangeScan_6]\n" +
+				"[  └─TableRangeScan_6 3333.33 cop[tikv] table:t5 range:(1,+inf], keep order:false, stats:pseudo]\n" +
+				"[└─Foreign_Key_Check_9 0.00 root table:t6, index:idx_id2 foreign_key:fk_2, check_not_exist]\n" +
```

Comment:
do not use hard code to compare the expected result, since the cost model / explain detail change need to refine which is time-consuming.  use explain-test for quick result recording if any